File: Application/app/src/main/java/com/example/sage/foodsharebook/Ingredients/IngredientMeasureActivity.java
package com.example.sage.foodsharebook.Ingredients;

import android.app.Activity;
import android.content.Intent;
import android.support.v7.app.AppCompatActivity;
import android.os.Bundle;
import android.util.Log;
import android.view.View;
import android.widget.Button;
import android.widget.EditText;
import android.widget.ImageView;
import android.widget.RadioButton;
import android.widget.RadioGroup;
import android.widget.TextView;
import android.widget.Toast;

import static com.example.sage.foodsharebook.Config.Constants.*;

import com.bumptech.glide.Glide;
import com.bumptech.glide.load.engine.DiskCacheStrategy;
import com.example.sage.foodsharebook.R;
import com.example.sage.foodsharebook.models.DishIngredient;

import java.util.ArrayList;
import java.util.List;

public class IngredientMeasureActivity extends AppCompatActivity {
    private static String TAG = "IngredientMeasureActivity";
    private TextView tv_ingredient_name;
    private EditText et_quantity;
    private ImageView iv_ingredient_image;
    private RadioGroup rg_measures;
    private Button btn_add_ingredient;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_ingredient_measure);

        iv_ingredient_image = findViewById(R.id.iv_ingredient_pic);
        tv_ingredient_name = findViewById(R.id.tv_ingredient_name);
        et_quantity = findViewById(R.id.et_measure_quantity);
        rg_measures = findViewById(R.id.rg_measures);
        btn_add_ingredient = findViewById(R.id.btn_add_ingredient);


        final Intent intent = getIntent();
        final String ingredient_name = intent.getStringExtra(INGREDIENT_NAME);
        final String ingredient_image = intent.getStringExtra(INGREDIENT_IMAGE);
        String measuresString = intent.getStringExtra(INGREDIENT_MEASURES);
        if(!measuresString.isEmpty()){
            List<Integer> measures = new ArrayList<>();
            for(String measure : measuresString.split(",")){
                measures.add(Integer.parseInt(measure));
            }
            makeRadioButtons(measures, rg_measures);
        }


        Glide.with(getApplicationContext())
                .load(ingredient_image)
                .centerCrop()
                .crossFade()
                .diskCacheStrategy(DiskCacheStrategy.ALL)
                .into(iv_ingredient_image);
        tv_ingredient_name.setText(ingredient_name);

        btn_add_ingredient.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View view) {
                if(et_quantity.getText().toString().isEmpty() || rg_measures.getCheckedRadioButtonId() == -1 ){
                    Toast.makeText(getApplicationContext(), "Fields missing", Toast.LENGTH_SHORT).show();
                } else {
                    int measure_id = (int)findViewById(rg_measures.getCheckedRadioButtonId()).getTag();
                    Log.i(TAG, "Mostrando ingrediente creado");
                    DishIngredient dishIngredient = new DishIngredient(
                      intent.getIntExtra(INGREDIENT_ID,-1),
                      ingredient_name,
                      ingredient_image,
                      measure_id,
                      Float.parseFloat(et_quantity.getText().toString().trim())
                    );
                    Intent intent = new Intent();
                    intent.putExtra(INGREDIENT_NAME, dishIngredient.getIngredientName());
                    intent.putExtra(INGREDIENT_IMAGE, dishIngredient.getIngredientImage());
                    intent.putExtra(QUANTITY, dishIngredient.getQuantity());
                    intent.putExtra(MEASURE_ID, dishIngredient.getMeasureId());
                    intent.putExtra(INGREDIENT_ID, dishIngredient.getIngredientId());
                    setResult(Activity.RESULT_OK, intent);
                    finish();
                }
            }
        });

    }

    private void makeRadioButtons(List<Integer> measures, RadioGroup radioGroup ){
        int i = 0;
        for(Integer measure : measures){
            RadioButton rb = new RadioButton(this);
            rb.setText(measureString(measure));
            rb.setTag(measure);
            radioGroup.addView(rb);
            i++;
        }
    }

    private String measureString(int measure){
        String result = "";
        switch (measure){
            case 1: result = "Piezas";
            break;
            case 2: result = "Gramos";
            break;
            case 3: result = "Kilogramos";
            break;
            case 4: result = "Mililitros";
            break;
            case 5: result = "Litros";
            break;
            case 6: result = "Cucharadas";
            break;
            case 7: result = "Tazas";
            break;
            case 8: result = "Rebanadas";
            break;
            case 9: result = "Pieza";
            break;
            default: result = "";
        }
        return result;
    }
}

File: Application/app/src/main/java/com/example/sage/foodsharebook/Ingredients/IngredientMeasureActivity.kt
package com.example.sage.foodsharebook.Ingredients

import android.app.Activity
import android.content.Intent
import android.support.v7.app.AppCompatActivity
import android.os.Bundle
import android.util.Log
import android.view.View
import android.widget.Button
import android.widget.EditText
import android.widget.ImageView
import android.widget.RadioButton
import android.widget.RadioGroup
import android.widget.TextView
import android.widget.Toast

import com.example.sage.foodsharebook.Config.Constants.*

import com.bumptech.glide.Glide
import com.bumptech.glide.load.engine.DiskCacheStrategy
import com.example.sage.foodsharebook.R
import com.example.sage.foodsharebook.models.DishIngredient

import java.util.ArrayList

class IngredientMeasureActivity : AppCompatActivity() {
    private var tv_ingredient_name: TextView? = null
    private var et_quantity: EditText? = null
    private var iv_ingredient_image: ImageView? = null
    private var rg_measures: RadioGroup? = null
    private var btn_add_ingredient: Button? = null

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_ingredient_measure)

        iv_ingredient_image = findViewById(R.id.iv_ingredient_pic)
        tv_ingredient_name = findViewById(R.id.tv_ingredient_name)
        et_quantity = findViewById(R.id.et_measure_quantity)
        rg_measures = findViewById(R.id.rg_measures)
        btn_add_ingredient = findViewById(R.id.btn_add_ingredient)


        val intent = intent
        val ingredient_name = intent.getStringExtra(INGREDIENT_NAME)
        val ingredient_image = intent.getStringExtra(INGREDIENT_IMAGE)
        val measuresString = intent.getStringExtra(INGREDIENT_MEASURES)
        if (!measuresString.isEmpty()) {
            val measures = ArrayList<Int>()
            for (measure in measuresString.split(",".toRegex()).dropLastWhile({ it.isEmpty() }).toTypedArray()) {
                measures.add(Integer.parseInt(measure))
            }
            makeRadioButtons(measures, rg_measures)
        }


        Glide.with(applicationContext)
                .load(ingredient_image)
                .centerCrop()
                .crossFade()
                .diskCacheStrategy(DiskCacheStrategy.ALL)
                .into(iv_ingredient_image!!)
        tv_ingredient_name!!.text = ingredient_name

        btn_add_ingredient!!.setOnClickListener {
            if (et_quantity!!.text.toString().isEmpty() || rg_measures!!.checkedRadioButtonId == -1) {
                Toast.makeText(applicationContext, "Fields missing", Toast.LENGTH_SHORT).show()
            } else {
                val measure_id = findViewById<View>(rg_measures!!.checkedRadioButtonId).tag as Int
                Log.i(TAG, "Mostrando ingrediente creado")
                val dishIngredient = DishIngredient(
                        intent.getIntExtra(INGREDIENT_ID, -1),
                        ingredient_name,
                        ingredient_image,
                        measure_id,
                        java.lang.Float.parseFloat(et_quantity!!.text.toString().trim { it <= ' ' })
                )
                val intent = Intent()
                intent.putExtra(INGREDIENT_NAME, dishIngredient.ingredientName)
                intent.putExtra(INGREDIENT_IMAGE, dishIngredient.ingredientImage)
                intent.putExtra(QUANTITY, dishIngredient.quantity)
                intent.putExtra(MEASURE_ID, dishIngredient.measureId)
                intent.putExtra(INGREDIENT_ID, dishIngredient.ingredientId)
                setResult(Activity.RESULT_OK, intent)
                finish()
            }
        }

    }

    private fun makeRadioButtons(measures: List<Int>, radioGroup: RadioGroup?) {
        var i = 0
        for (measure in measures) {
            val rb = RadioButton(this)
            rb.text = measureString(measure)
            rb.tag = measure
            radioGroup!!.addView(rb)
            i++
        }
    }

    private fun measureString(measure: Int): String {
        var result = ""
        when (measure) {
            1 -> result = "Piezas"
            2 -> result = "Gramos"
            3 -> result = "Kilogramos"
            4 -> result = "Mililitros"
            5 -> result = "Litros"
            6 -> result = "Cucharadas"
            7 -> result = "Tazas"
            8 -> result = "Rebanadas"
            9 -> result = "Pieza"
            else -> result = ""
        }
        return result
    }

    companion object {
        private val TAG = "IngredientMeasureActivity"
    }
}
